packages/protocol/contracts/dapis/SamplePriceDataFeed.sol

```diff
@@ -19,6 +37,170 @@ contract SamplePriceDataFeed {
     latestDapiId = dapiId;
   }
 
+  function addTemplate(
+    bytes16 _dapiId,
+    bytes32 _templateId,
+    address _designatedWallet
+  ) external onlyAdminOrMetaAdmin cooldownTimeEnded {
+    // PriceDataFeed fetches the current dAPI parameters from RrpDapiServer
+    (
+      uint256 noResponsesToReduce,
+      uint256 toleranceInPercentages,
+      uint256 requesterIndex,
+      bytes32[] memory templateIds,
+      address[] memory designatedWallets,
+      address reduceAddress,
+      bytes4 reduceFunctionId,
+      address requestIndexResetter
+    ) = rrpDapiServer.getDapi(_dapiId);
+    // TODO: check dAPI was found?
+
+    // Adds templateId to dAPI.templateIds (iterating over because array push() is only available in storage arrays)
+    // Also add _designatedWallet to dAPI.designatedWallets
+    bytes32[] memory newTemplateIds = new bytes32[](templateIds.length + 1);
+    address[] memory newDesignatedWallets = new address[](designatedWallets.length + 1);
+    for (uint256 i = 0; i < templateIds.length; i++) {
+      newTemplateIds[i] = templateIds[i];
+      newDesignatedWallets[i] = designatedWallets[i];
+    }
+    newTemplateIds[newTemplateIds.length - 1] = _templateId;
+    newDesignatedWallets[newDesignatedWallets.length - 1] = _designatedWallet;
+
+    // Calls RrpDapiServer to create the new dAPI
+    bytes16 newDapiId =
+      rrpDapiServer.registerDapi(
+        noResponsesToReduce,
+        toleranceInPercentages,
+        requesterIndex,
+        newTemplateIds,
+        newDesignatedWallets,
+        reduceAddress,
+        reduceFunctionId,
+        requestIndexResetter
+      );
+
+    // Update dapiLastUpdated timestamp
+    adminToDapiLastUpdated[msg.sender] = uint64(block.timestamp);
+
+    // Record the returned dapiId
+    latestDapiId = newDapiId;
+
+    // TODO: publish TemplateAdded event?
```

Comment:
What do you see the benefit being?

Reply:
(Actually wondering, not being smarmy haha)

Reply:
Basically I see this aggregator contract being used by possible many client apps and they might want to know what changed on the underlying dAPI everytime an admin calls any of these functions that cause a mutation on the dAPI. Not sure tho and that's why there is a TODO :smile:

Reply:
That makes sense